spdx/v2/v2_2/json/json_test.go

```diff
@@ -71,6 +71,11 @@ func TestLoad(t *testing.T) {
 func Test_Write(t *testing.T) {
 	want := example.Copy()
 
+	// we always output FilesAnalyzed, even though we handle reading files where it is omitted
```

Comment:
the weird part of filesAnalyzed is that if it is omitted technically the correct value is "true" but really no one does it this way... which is kind of cringe, because the intention is usually `filesAnalyzed=false`... if we actually did a validation check when reading, likely >90% of SPDX documents will fail since they assume omission is default false based on the programming language defaults, and thus they don't include `VerificationCode` which technically is required if `filesAnalyzed=true`. 

I see trade offs both ways, but since we do require user to signify their intention and the use of the library is not asking users to provide a mandatory field, a golang user will think that it defaults to false. My thought is to perhaps maybe flip the `IsFilesAnalyzedTagPresent` to `IsFilesAnalyzedTagOmitted`, or make the default value to true... i don't quite want to break the interface. This may create a experience a bit more aligned with golang

This however, doesn't fix the issue of all the input SBOMs having their filesAnalyzed intent misinterpreted, which technically isn't an issue for the library... but i think it will break enough of the ecosystem that we have to consider it.

Reply:
i thought about it a bit more... maybe what we can do is that during reading, is that if we see no `VerificationCode` we can set `filesAnalyzed=false`.. wdyt @kzantow ?

Reply:
the more i look into this, the worse it gets... since `filesAnalyzed=false` is also an indication that a package doesn't contain any files, this messes things up.. 

I think my new thoughts around this is to just output it `IsFilesAnalyzedTagPresent=true`, and otherwise omit it and not assert anything about its intended value but to parse it as is.. We think we will need to support this bad behavior as accounting for a deficiency of the spec.